module-2/backend/week_one.md

```diff
@@ -4,31 +4,97 @@ Fork this respository. Answer the questions to the best of your ability. Try to
+
+*instance variables*
+
+Given the following block of code, how would I pass an instance variable `count` with a value of `1` to my `index.erb` template? 
+
+In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
   
   ```ruby
   get '/horses' do
+    **@count = 1**
+    **@name = "Mr. Ed"
     erb :index
   end
-  ```
-
-8. In the same code block, how would I pass a local variable `name` with a value of `Mr. Ed` to the view?
-9. What's the purpose of ERB?
-10. Why do I need a development AND test database?
-11. What's responsive design?
-12. What is CRUD and why is it important?
-13. What does HTTP stand for? 
-14. What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
-15. What's an ORM?
-16. What's the most commonly used ORM in ruby (Sinatra & Rails)?
-17. Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
-18. What's a migration? 
-19. When you create a migration, does it automatically modify your database?
-20. How does a model relate to a database?
-21. What's the difference between agile workflow and waterfall method?
-22. What is the difference between `#new` and `#create`?
+```
+
+What's the purpose of ERB?
+
+*the erb file type allows us to write ruby code in the html file to render elements on the page*
+
+Why do I need a development AND test database?
+
+*To separate out the two different databases. In development, data is usually not important and can be easily deleted or experimented with. With test, data fidelity is more important for test accuracy.*
+
+What's responsive design?
+
+*???*
+
+What is CRUD and why is it important?
+
+*Create, Read, Update, Delete -- it's important because this functionality is how the user interacts with the database (and therefore the application)*
+
+What does HTTP stand for?
+
+*Hypertext Transfer Protocol*
+
+What are the two ways to interpolate Ruby in an ERB view template? What's the difference between these two ways?
+
+*I only know <% %>*
+
+What's an ORM?
+
+*???*
+
+What's the most commonly used ORM in ruby (Sinatra & Rails)?
+
+*???*
+
+Let's say we have an application with restaurants. There are seven verb + path combinations necessary to provide full CRUD functionality for our restaurant application. List each of the seven combinations, and explain what each is for.
+
+* *Create -- get/post -- get the page with the form and post the parameters from the form when the user clicks submit*
+* *Read -- get -- get the page*
+* *Update -- get/put -- get the page with the form and the current values populated in each field, and put the parameters from the form when the user clicks submit*
+* *Delete -- delete -- when the user clicks the button, delete the specific entry associated with that button*

```

Comment:
These are the HTTP verbs, but the seven verb/path combinations are as follows:

GET /restaurants/new
GET /restaurants
POST /restaurants
GET /restaurants/:id/edit
GET /restauratns/:id
PUT /restaurants/:id
DELETE /restaurants/:id

Recommend reviewing so that you can recreate the above and know what each does.